internal/model/puller.go

```diff
@@ -367,13 +398,12 @@ func (p *Puller) pullerIteration(ignores *ignore.Matcher) int {
 	// Wait for the finisherChan to finish.
 	doneWg.Wait()
 
-	for i := range deletions {
-		deletion := deletions[len(deletions)-i-1]
-		if deletion.IsDirectory() {
-			p.deleteDir(deletion)
-		} else {
-			p.deleteFile(deletion)
-		}
+	for _, file := range fileDeletions {
+		p.deleteFile(file)
```

Comment:
I was thinking that we don't actually need the FileInfo at all to do the delete, but we do (since deleteFile calls updateLocal), so ignore that comment.